PowerPointLabs/PowerPointLabs/PositionsLab/PositionsPaneWPF.xaml.cs

```diff
@@ -1531,8 +1573,38 @@ private void ExecuteDistributeGrid(int rowLength, int colLength)
             {
                 if (simulatedShapes != null)
                 {
-                    simulatedShapes.Delete();
-                    GC.Collect();
+                    try
+                    {
+                        simulatedShapes.Delete();
+                        GC.Collect();
+                    }
+                    // Catch corrupted shapes
+                    catch (System.Runtime.InteropServices.COMException)
+                    {
+                        // Remove all simulated shapes manually
+                        for (int i = 0; i < simulatedShapes.Count; i++)
+                        {
+                            // This method to remove duplicated shapes might fail for non-corrupted shapes when mixing good/bad shapes
+                            if (this.GetCurrentSlide().Shapes[1].Name.Contains("_Copy"))
```

Comment:
Will this unintentionally delete some shapes if its name has _Copy? (Even when it should not be affected by this operation)